app/main.py

```diff
@@ -1,2 +1,82 @@
+from dataclasses import dataclass, field
+from typing import Any
+from app.point import Point
+
+
+@dataclass
+class Node:
+    key: Any
+    hash_val: int
+    value: Any
+
+
+@dataclass
 class Dictionary:
-    pass
+    initial_capacity = 8
+    load_factor = 2 / 3
+    size = 0
+    hash_table: list[list[Node]] = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.hash_table = [None] * self.initial_capacity
+
+    def line_stop(self) -> int:
+        return int(self.initial_capacity * self.load_factor)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_val = hash(key)
+        index = hash_val % self.initial_capacity
+        if self.hash_table[index] is None:
+            self.hash_table[index] = [Node(key, hash_val, value)]
+
+        else:
+            for node in self.hash_table[index]:
+                if node.hash_val == hash_val and node.key == key:
+                    node.value = value
+                    return
```

Comment:
this is not correct logic,  `self.hash_table[index] ` must contain no more than one node